src/promptflow-devkit/promptflow/orchestrator/_chat_group_orchestrator.py

```diff
@@ -0,0 +1,142 @@
+from promptflow.executor._result import LineResult
+from promptflow.storage import AbstractRunStorage
+from promptflow.batch._batch_inputs_processor import BatchInputsProcessor
+from promptflow._utils.execution_utils import apply_default_value_for_input
+from promptflow._proxy._proxy_factory import ProxyFactory
+
+
+class ChatGroupOrchestrator:
+    def __init__(
+        self,
+        chat_group_roles: List[ChatGroupRole],
+        max_turn: Optional[int] = None,
+        storage: Optional[AbstractRunStorage] = None,
+        max_lines_count: Optional[int] = None,
+        **kwargs
+    ):
+        self._storage = storage
+        self._max_turn = max_turn
+        self._chat_group_roles = chat_group_roles
+        self._max_lines_count = max_lines_count
+
+        self._executor_proxies: List[AbstractExecutorProxy] = self._create_executor_proxy(**kwargs)
+
+    @classmethod
+    def create(
+        cls,
+        chat_group_roles: List[ChatGroupRole],
+        max_turn: Optional[int] = None,
+        storage: Optional[AbstractRunStorage] = None,
+        max_lines_count: Optional[int] = None,
+    ) -> "ChatGroupOrchestrator":
+
+        for chat_role in chat_group_roles:
+            chat_role.working_dir = Flow._resolve_working_dir(chat_role.flow_file, chat_role.working_dir)
+            chat_role.flow = Flow.from_yaml(chat_role.flow_file, working_dir=chat_role.working_dir)
+
+        return ChatGroupOrchestrator(chat_group_roles, max_turn, storage, max_lines_count)
+
+    def _create_executor_proxy(self, **kwargs) -> List[AbstractExecutorProxy]:
+        executor_proxy_list = []
+        executor_proxy_factory = ProxyFactory()
+        for chat_role in self._chat_group_roles:
+            executor_proxy = executor_proxy_factory.create_executor_proxy(
+                flow_file=chat_role.flow_file,
+                working_dir=chat_role.working_dir,
+                connections=chat_role.connections,
+                storage=self._storage,
+                language=self._check_language_from_yaml(chat_role),
+                **kwargs
+            )
+            executor_proxy_list.append(executor_proxy)
+        return executor_proxy_list
+
+    async def destroy(self):
+        for executor_proxy in self._executor_proxies:
+            await executor_proxy.destroy()
+
+    async def _schedule_runs(
+            self,
+            line_index: int,
+            inputs: Mapping[str, Any] = None,
+            run_id: str = None,
+            ) -> LineResult:
+
+        """schedule runs for a line, submit roleA and format its output as roleB's input.
+        Then submit roleB until the max_turn.
+        """
+
+        outputs: dict = {}
+        aggregation_inputs: dict = {}
+        current_line_result: LineResult = None
+
+        total_roles = len(self._chat_group_roles)
+        conversation_history: List[Mapping[str, Any]] = []
+        batch_inputs = self._process_batch_inputs(inputs)
+        for turn in range(self._max_turn):
```

Comment:
May need more logs about current turn information (and for loop break) for debug usage.